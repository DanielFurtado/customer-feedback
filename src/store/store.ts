import { configureStore } from '@reduxjs/toolkit';
import ratingReducer from './rating/rating.reducer';

export const store = configureStore({
  reducer: {
    feedbackList: ratingReducer
  }
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
