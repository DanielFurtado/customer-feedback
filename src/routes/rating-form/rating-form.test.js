import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import RatingForm from './rating-form';
import {BrowserRouter as Router} from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../store/store';

describe(('Rating Form'), () => {
  it('renders', () => {
    let component = renderer.create(
      <Provider store={store}>
        <Router>
          <RatingForm />
        </Router>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});