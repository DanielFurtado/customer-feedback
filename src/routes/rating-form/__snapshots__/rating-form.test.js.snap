// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Form renders 1`] = `
<main
  className="sc-gswNZR bODSCs"
>
  <form
    className="sc-hLBbgP gCMYbT"
    noValidate={true}
  >
    <h2>
      Feedback Form
    </h2>
    <div
      className="sc-eDvSVe eqITwC"
    >
      <div
        className="sc-jSUZER iJjbrW"
      >
        <div
          className="Form--group"
        >
          <label
            htmlFor="name"
          >
            Name *
          </label>
          <input
            aria-invalid="false"
            aria-label="Name"
            id="name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Name"
            type="text"
          />
        </div>
        <div
          className="Form--group"
        >
          <label
            htmlFor="email"
          >
            Email *
          </label>
          <input
            aria-invalid="false"
            aria-label="Email"
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email"
            type="text"
          />
        </div>
        <div
          className="Form--group"
        >
          <label
            htmlFor="rating"
          >
            Rating *
          </label>
          <input
            aria-invalid="false"
            aria-label="Rating"
            id="rating"
            max="5"
            min="1"
            name="rating"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Rating (1-5 stars)"
            type="number"
          />
        </div>
      </div>
      <div
        className="sc-jSUZER iJjbrW"
      >
        <div
          className="sc-gKPRtg dgIdrf"
        >
          <label
            htmlFor="comment"
          >
            Comment *
          </label>
          <textarea
            aria-invalid="false"
            aria-label="Comment"
            id="comment"
            name="comment"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Comment"
          />
        </div>
      </div>
    </div>
    <div
      className="sc-iBYQkv dkpTLM"
    >
      <button
        className="sc-dkrFOg dxOIDC"
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</main>
`;
